packages/ra-data-simple-rest/src/index.ts

```diff
@@ -156,10 +156,13 @@ export default (
     updateMany: (resource, params) =>
         Promise.all(
             params.ids.map(id =>
-                httpClient(`${apiUrl}/${resource}/${id}`, {
-                    method: 'PUT',
-                    body: JSON.stringify(params.data),
-                })
+                httpClient(
+                    `${apiUrl}/${resource}/${encodeURIComponent(typeof id === 'object' ? id.id : String(id))}`,
```

Comment:
I don't understand why you test that `id` may be an object. The interface doesn't allow for that. Same for `deleteMany`.

Reply:
An existing deleteMany test had that as input so I assumed that was the format. Fixing.